Symfony/CS/Console/Command/FixCommand.php

```diff
@@ -545,7 +548,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $this->listErrors($output, 'linting after fixing', $lintErrors);
         }
 
-        return !$resolver->isDryRun() || empty($changed) ? 0 : 3;
+        $exitStatus = 0;
+
+        if ($resolver->isDryRun()) {
+            if (!empty($invalidErrors)) {
+                $exitStatus |= self::EXIT_STATUS_FLAG_HAS_INVALID_FILES;
+            }
```

Comment:
Indeed. Could you send a PR for that ? (but keep `EXIT_STATUS_FLAG_HAS_CHANGED_FILES` only in dry-run)